exporters/trace/ocagent/src/main/java/io/opencensus/exporter/trace/ocagent/OcAgentNodeUtils.java

```diff
@@ -112,72 +106,61 @@ static ServiceInfo getServiceInfo(String serviceName) {
     return ServiceInfo.newBuilder().setName(serviceName).build();
   }
 
-  /*
-   * Creates an attribute map with the given MonitoredResource.
-   * If the given resource is not null, the attribute map contains exactly two entries:
-   *
-   * OPENCENSUS_SOURCE_TYPE:
-   *   A string that describes the type of the resource prefixed by a domain namespace,
-   *   e.g. “kubernetes.io/container”.
-   * OPENCENSUS_SOURCE_LABELS:
-   *   A comma-separated list of labels describing the source in more detail,
-   *   e.g. “key1=val1,key2=val2”. The allowed character set is appropriately constrained.
-   */
-  // TODO: update the resource attributes once we have an agreement on the resource specs:
-  // https://github.com/census-instrumentation/opencensus-specs/pull/162.
+  @Nullable
+  static Resource getDefaultResourceProto() {
+    return toResourceProto(RESOURCE);
+  }
+
+  // Converts a MonitoredResource to a Resource proto.
+  // TODO: update this method to use the Java Resource API once auto-detection is implemented.
+  @Nullable
   @VisibleForTesting
-  static Map<String, String> getAttributeMap(@Nullable MonitoredResource resource) {
+  static Resource toResourceProto(@Nullable MonitoredResource resource) {
     if (resource == null) {
-      return Collections.emptyMap();
+      return null;
     } else {
-      Map<String, String> resourceAttributes = new HashMap<String, String>();
-      resourceAttributes.put(RESOURCE_TYPE_ATTRIBUTE_KEY, resource.getResourceType().name());
-      resourceAttributes.put(RESOURCE_LABEL_ATTRIBUTE_KEY, getConcatenatedResourceLabels(resource));
-      return resourceAttributes;
+      Resource.Builder resourceProtoBuilder = Resource.newBuilder();
+      resourceProtoBuilder.setType(resource.getResourceType().name());
+      putResourceLabels(resource, resourceProtoBuilder);
+      return resourceProtoBuilder.build();
     }
   }
 
-  // Encodes the attributes of MonitoredResource into a comma-separated list of labels.
-  // For example "aws_account=account1,instance_id=instance1,region=us-east-2".
-  private static String getConcatenatedResourceLabels(MonitoredResource resource) {
-    StringBuilder resourceLabels = new StringBuilder();
+  // Puts the attributes of MonitoredResource to ResourceProto.
+  private static void putResourceLabels(
```

Comment:
We should probably change that package the auto-discovery package to use the Resource but in a different PR.

Reply:
Agree, filed https://github.com/census-instrumentation/opencensus-java/issues/1583 to keep track.